src/Planner/PlannerJoinTree.cpp

```diff
@@ -208,6 +208,9 @@ bool applyTrivialCountIfPossible(
     if (row_policy_filter)
         return {};
 
+    if (select_query_info.additional_filter_ast)
```

Comment:
What about additional result filters?

Reply:
The issue is that trivial count optimization ignores additional filer and creates a reading step with invalid count result.
Result filters does not have this issue: it is an expression on top of query result.